packages/location-management/src/components/LocationTree/index.tsx

```diff
@@ -1,110 +1,92 @@
 import { SearchOutlined } from '@ant-design/icons';
-import reducerRegistry from '@onaio/redux-reducer-registry';
-import {
-  getLocationTreeState,
-  reducerName,
-  setLocationTreeState,
-  reducer,
-} from '../../ducks/location-hierarchy';
 import { AntTreeProps } from '../LocationUnitList';
 import './tree.css';
 import { ParsedHierarchyNode } from '../../ducks/locationHierarchy/types';
-import { getHierarchyNodeFromArray } from '../../ducks/locationHierarchy/utils';
 import { useTranslation } from '../../mls';
+import TreeModel from 'tree-model';
 import { Key } from 'rc-tree/lib/interface';
-reducerRegistry.register(reducerName, reducer);
+
+/** helper type, shortened form */
+export type TreeNode = TreeModel.Node<ParsedHierarchyNode>;
+
+interface SelectCallbackInfo {
+  event: 'select';
+  selected: boolean;
+  node: TreeNode;
+  selectedNodes: TreeNode;
+  nativeEvent: MouseEvent;
+}
 
 interface TreeProp {
-  data: ParsedHierarchyNode[];
-  OnItemClick: (item: ParsedHierarchyNode) => void;
+  data: TreeNode[];
+  selectedNode: TreeNode | undefined;
+  onSelect: (item: TreeNode | undefined) => void;
+}
+
+interface ExpandCallbackInfo {
+  node: TreeNode;
+  expanded: boolean;
+  nativeEvent: MouseEvent;
 }
 
 export const Tree: React.FC<TreeProp> = (props: TreeProp) => {
-  const { data, OnItemClick } = props;
-  const dispatch = useDispatch();
+  const { data, onSelect, selectedNode } = props;
   const { t } = useTranslation();
 
   const [expandedKeys, setExpandedKeys] = useState<Key[]>([]);
   const [searchValue, setSearchValue] = useState<string>('');
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const filterData: ParsedHierarchyNode[] = [];
-  const locationTreeState = useSelector((state) => getLocationTreeState(state));
-
-  /**
-   * Function to to expand tree
-   *
-   * @param {string} value - value to be searched and expanded
-   */
-  function expandTree(value: string) {
-    const expandedKeys = filterData
-      .map((item) =>
-        value.length && item.label.toLocaleLowerCase().indexOf(value.toLowerCase()) > -1
-          ? getParentKey(item.id, item.parent as string, filterData)
-          : null
-      )
-      .filter((item, i, self) => item && self.indexOf(item) === i);
-    setExpandedKeys(expandedKeys as string[]);
-    dispatch(
-      setLocationTreeState({
-        keys: expandedKeys as React.Key[],
-        node: locationTreeState?.node as ParsedHierarchyNode,
-      })
-    );
-  }
+  const [searchExpandedKeys, setSearchExpandedKeys] = useState<Key[]>([]);
 
   useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    if (locationTreeState?.node) {
-      const node = getHierarchyNodeFromArray(filterData, locationTreeState.node.id);
-      if (node) {
-        onExpand(locationTreeState.keys);
-        OnItemClick(node);
-      }
+    const updateExpandedKeys = (keys: Key[]) => setExpandedKeys([...expandedKeys, ...keys]);
```

Comment:
can this take expandedKeys as an arg?

Reply:
yeah, though I think making use of closure here makes using the util function more pleasant to work with, though again I can be convinced otherwise.

Reply:
No strong opinion, just trying to keep the scope clean and functions pure